### PR TITLE
Name resolution errors e2e tests

### PR DESCRIPTION
**Pull request by JHodorHodor**
All new tests have `@Ignore` annotation! I started with a branch of the parser tests, so probably it is the best to compare changes with it.

### CODE REVIEW

**Comment by ignacy123** [comment]
I think it is also worth adding the following test case:
```
x = 15
zm x: Liczba = 17
```
to ensure we don't mess up the order somewhere. Maybe the same idea can be used for other scenarios?

File: app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt [hunk 1]
```diff
@@ -1,5 +1,246 @@
...
+
+    // I'm not 100% sure if the next two tests make sense
+    @Ignore
+    @Test
+    fun `test using functions in assignment`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        czynność g() { }
+                        zm x: Liczba = x
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt:224]: I think it should be `zm x: Liczba = g`

File: app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt [hunk 2]
```diff
@@ -1,5 +1,246 @@
...
+                        zm x: Czy = prawda
+                        zwróć x()
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.VariableIsNotCallable::class
+        )
+    }
+
+    // I'm not 100% sure if the next two tests make sense
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt:216]: In my opinion, they make sense - our language doesn't support using functions as variables at this point.

File: app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt [hunk 3]
```diff
@@ -1,5 +1,246 @@
...
+            """
+                    czynność f() -> Czy {
+                        czynność g() { }
+                        zwróć (g == 17)
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.FunctionIsNotVariable::class
+        )
+    }
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt:245]: Could you please add some tests for function parameters? We could test, for example, name conflicts (between two parameters and between a parameter and a variable), treating parameters as callables, using parameters outside function scope, and maybe using undefined variables/functions as default parameter values:
```
czynność f(x: Liczba = y) {}
czynność f(x: Liczba = f()) {}
```

File: app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt [hunk 4]
```diff
@@ -1,5 +1,246 @@
...
+    fun `test using functions in conditions`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Czy {
+                        czynność g() { }
+                        zwróć (g == 17)
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.FunctionIsNotVariable::class
```
Comment (kamgal00) [inline app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt:243]: This test is probably redundant - the previous one already tests if `g` is a correct expression

**Review by kamgal00 — CHANGES_REQUESTED** [review]
Testy są dobrze napisane i rozpatrują wszystkie przypadki, brakuje mi jedynie kilku testów dla parametrów funkcji.

**Review by kamgal00 — APPROVED** [review]

**Pull request merged**